src/main/java/com/input/Configuration.java

```diff
@@ -194,22 +200,24 @@ private boolean parseEventNodes(NodeList eventNodeList, HashMap<String, DataID>
                     if (dataNode.getNodeType() == curNode.ELEMENT_NODE){
                         Element elemData = (Element) dataNode;
 
+                        if (!elemData.hasAttribute("name")){                                   // Check if data element has no name
+                            ErrorHandler.logError("Name of data element is missing");
+                            return false;
+                        }
                         String dataName = elemData.getAttribute("name");    // Get Event object attributes out of node and temp hashmap
```

Comment:
I just remembered while writing my own code, you should make sure to trim() the attribute that you collect from the config file, just in case someone accidentally types a meaningless space, that probably shouldn't stop execution. This goes for any attribute that you collect from the config file!